src/lib.rs

```diff
@@ -386,11 +405,65 @@ pub struct SentryCredential {
     pub host: String,
     pub project_id: String,
 }
+
 pub struct Sentry {
+    settings: Settings,
+    worker: Arc<SingleWorker<Event, SentryCredential>>,
+}
+
+#[derive(Debug, PartialEq)]
+pub struct Settings {
     server_name: String,
     release: String,
     environment: String,
-    worker: Arc<SingleWorker<Event, SentryCredential>>,
+    device: Device
+}
+
+pub struct SettingsBuilder {
+    server_name: Option<String>,
+    release: Option<String>,
+    environment: Option<String>,
+    device: Option<Device>
+}
+
+impl SettingsBuilder {
+    pub fn new() -> SettingsBuilder {
```

Comment:
The builder here feels appropriate for a couple reasons. It lets us add new values to settings in the future without changing the existing public api, and it lets us have Settings objects that don't contain options so they're always fully-specified and easy to work with. Plus it'll be [familiar](https://aturon.github.io/ownership/builders.html) to rust users from other libraries that use multiple configurable and/or optional settings. (And personally I like the ergonomics of a builder over creating a mutable default config and modifying it over several statements). That's my two cents at least, we can go the other route if you prefer.